plutus-core/src/Language/PlutusCore/Builtins.hs

```diff
@@ -0,0 +1,371 @@
+            ((<) @Integer)
+            (runCostingFunTwoArguments . paramLessThanInteger)
+    toBuiltinMeaning LessThanEqInteger =
+        toStaticBuiltinMeaning
+            ((<=) @Integer)
+            (runCostingFunTwoArguments . paramLessThanEqInteger)
+    toBuiltinMeaning GreaterThanInteger =
+        toStaticBuiltinMeaning
+            ((>) @Integer)
+            (runCostingFunTwoArguments . paramGreaterThanInteger)
+    toBuiltinMeaning GreaterThanEqInteger =
+        toStaticBuiltinMeaning
+            ((>=) @Integer)
+            (runCostingFunTwoArguments . paramGreaterThanEqInteger)
+    toBuiltinMeaning EqInteger =
+        toStaticBuiltinMeaning
+            ((==) @Integer)
+            (runCostingFunTwoArguments . paramEqInteger)
+    toBuiltinMeaning Concatenate =
+        toStaticBuiltinMeaning
+            BS.append
+            (runCostingFunTwoArguments . paramConcatenate)
+    toBuiltinMeaning TakeByteString =
+        toStaticBuiltinMeaning
+            (BS.take . integerToInt)
+            (runCostingFunTwoArguments . paramTakeByteString)
+    toBuiltinMeaning DropByteString =
+        toStaticBuiltinMeaning
+            (BS.drop . integerToInt)
+            (runCostingFunTwoArguments . paramDropByteString)
+    toBuiltinMeaning SHA2 =
+        toStaticBuiltinMeaning
+            Hash.sha2
+            (runCostingFunOneArgument . paramSHA2)
+    toBuiltinMeaning SHA3 =
+        toStaticBuiltinMeaning
+            Hash.sha3
+            (runCostingFunOneArgument . paramSHA3)
+    toBuiltinMeaning VerifySignature =
+        toStaticBuiltinMeaning
+            (verifySignature @EvaluationResult)
+            (runCostingFunThreeArguments . paramVerifySignature)
+    toBuiltinMeaning EqByteString =
+        toStaticBuiltinMeaning
+            ((==) @BS.ByteString)
+            (runCostingFunTwoArguments . paramEqByteString)
+    toBuiltinMeaning LtByteString =
+        toStaticBuiltinMeaning
+            ((<) @BS.ByteString)
+            (runCostingFunTwoArguments . paramLtByteString)
+    toBuiltinMeaning GtByteString =
+        toStaticBuiltinMeaning
+            ((>) @BS.ByteString)
+            (runCostingFunTwoArguments . paramGtByteString)
+    toBuiltinMeaning IfThenElse =
+        toStaticBuiltinMeaning
+            ((\b x y -> if b then x else y) :: a ~ Opaque term (TyVarRep "a" 0) => Bool -> a -> a -> a)
+            (runCostingFunThreeArguments . paramIfThenElse)
+    toBuiltinMeaning CharToString =
+        toStaticBuiltinMeaning
+            (pure :: Char -> String)
+            mempty  -- TODO: budget.
+    toBuiltinMeaning Append =
+        toStaticBuiltinMeaning
+            ((++) :: String -> String -> String)
+            mempty  -- TODO: budget.
+    toBuiltinMeaning Trace =
+        toDynamicBuiltinMeaning
+            (\env -> unsafePerformIO . defaultFunDynTrace env)
+            mempty  -- TODO: budget.
+
+-- See Note [Stable encoding of PLC]
+instance Serialise DefaultFun where
```

Comment:
@kwxm 

> Just a reminder that for n >= 24, CBOR needs two bytes instead of one to encode n, so we want the commonest builtins at the front.

Attached your comment to the definition of `DefaultFun`.

@raduom wanna add anything regarding `Flat`?

Reply:
There may be something I do not understand. Why do we want the commonest builtins at the front? Does CBOR encode the data using a variable-length field, such that if you need to encode an n < 24, then it is more efficient? Flat does no such thing, it always takes the same space to encode n variants.

Reply:
Yeah, I think CBOR does small integers specially, so effectively a variable-length encoding. Given that we're hopefully not going to be using CBOR in production, I think we don't have to worry too much about it.

Reply:
> Given that we're hopefully not going to be using CBOR in production

Can we drop it completely then?

Reply:
> Can we drop it completely then?

I want to say we should keep it "just in case", but I don't have a good reason. Let me think about it a bit more in case there's something I'm missing, but maybe we could just drop it.

Reply:
> Can we drop it completely then?

I vote to keep it.  CBOR libraries seem to exist for a number of languages, but I don't know if there's _any_ implementation of Flat except for the Haskell one (the name's so generic that searching the internet is pretty hopeless).  I'm still tempted to write a CEK machine in OCaml just to see what happens, and CBOR would make it easy to load scripts in without a parser.

Reply:
Minor nitpick. When I switch to Flat I will probably scan the source for CBOR instances and replace those, so I vote to keep it as well.